Index: q.smt.generate.arc
===================================================================
--- q.smt.generate.arc	(revision 310)
+++ q.smt.generate.arc	(revision 316)
@@ -1388,6 +1388,14 @@
       .invoke r = gen_parameter_list( v_pars, false, "function" )
       .assign te_parm = r.result
       .assign parameters = te_parm.ParamBuffer
+.//-- 018:20140402 Add Start (saitou)
+    .if ( te_sync.IsInitFunction != TRUE )
+      .if ( "$r{parameters}" != "" )
+        .assign parameters = parameters + ","
+      .end if
+      .assign parameters = parameters + " 0 "
+    .end if
+.//-- 018:20140402 Add End (saitou)
       .assign parameter_OAL = te_parm.OALParamBuffer
     .end if
     .assign function_name = te_sync.intraface_method
