Index: fx_util.arc
===================================================================
--- fx_util.arc	(revision 318)
+++ fx_util.arc	(revision 320)
@@ -232,6 +232,7 @@
 	.end if
 ;
     .end if
+  .end if
 .end function
 .//
 .//----------------------------------------------------------------------------
@@ -352,7 +353,7 @@
       .assign te_class.InstanceMaxNoDefIncludeFile = includeFile
     .else
       .print "ERROR: Class ${className} not found.\n => mark_obj_as_fixed_inst_no_extend()"
-    .end
+    .end if
   .else
       .print "ERROR: Component ${domainName} not found.\n => mark_obj_as_fixed_inst_no_extend()"
   .end if
