Index: q.class.arc
===================================================================
--- q.class.arc	(revision 320)
+++ q.class.arc	(revision 321)
@@ -304,6 +304,9 @@
       .select any te_class related by te_c->TE_CLASS[R2064]
       .if ( ( not_empty te_sync ) or ( not_empty te_class ) )
 #include "${te_c.classes_file}.${te_file.hdr_file_ext}"
+.//-- 021:20140403 Add Start (nomura)
+#include "${te_c.Name}_mechs.${te_file.hdr_file_ext}"
+.//-- 021:20140403 Add End (nomura)
       .end if
     .end if
   .end if
