Index: fx_domain_facade.arc
===================================================================
--- fx_domain_facade.arc	(revision 326)
+++ fx_domain_facade.arc	(revision 337)
@@ -69,6 +69,10 @@
     ${base_arch_prefix.result}_callOut_Fp timerEmptyCalloutFP /* dummy */
 );
 
+.//-- 024:20140409 Add Start (nomura)
+void ${domainName}_DomainFacade_Execute(${base_arch_prefix.result}_threadInfo_t *);
+.//-- 024:20140409 Add End (nomura)
+    .//
 .end function
 .//
 .//============================================================================
