Index: fx_domain_facade.arc
===================================================================
--- fx_domain_facade.arc	(revision 337)
+++ fx_domain_facade.arc	(revision 338)
@@ -62,7 +62,7 @@
     .select any te_file from instances of TE_FILE;
     .assign domainName = te_c.Name
     .//
-void ${domainName}_DomainFacade_Initialize(
+extern void ${domainName}_DomainFacade_Initialize(
     ${base_arch_prefix.result}_waitForExternalEvents_Fp waitForExtEventFp,
     ${base_arch_prefix.result}_callOut_Fp eventEmptyCalloutFP,
     ${base_arch_prefix.result}_callOut_Fp nodeEmptyCalloutFP,
@@ -70,7 +70,7 @@
 );
 
 .//-- 024:20140409 Add Start (nomura)
-void ${domainName}_DomainFacade_Execute(${base_arch_prefix.result}_threadInfo_t *);
+extern void ${domainName}_DomainFacade_Execute(${base_arch_prefix.result}_threadInfo_t *);
 .//-- 024:20140409 Add End (nomura)
     .//
 .end function
