Index: t.class.extent.h
===================================================================
--- t.class.extent.h	(revision 338)
+++ t.class.extent.h	(revision 345)
@@ -25,7 +25,7 @@
 .//-- 010:20140224 Add Start (nomura)
 extern void ${te_class.GeneratedName}_FactoryInit();
 extern ${te_instance.handle} ${te_class.GeneratedName}_Create();
-extern void ${te_class.GeneratedName}_Delete();
+extern void ${te_class.GeneratedName}_Delete(${defaultPrefix}_iHandle_t);
 .//-- 010:20140224 Add End (nomura)
 
 .//-- 019:20140328 Add Start (nomura)
Index: q.class.factory.arc
===================================================================
--- q.class.factory.arc	(revision 338)
+++ q.class.factory.arc	(revision 345)
@@ -39,6 +39,10 @@
   .invoke r = GetFixedSizeClassExtentInfo( te_class )
   .assign extent = r.result
   .if ( gen_declaration )
+.//-- 010:20140410 Add Start (saitou) 
+    .invoke default_arch_prefix = fx_get_default_arch_prefix_name()
+    .assign defaultPrefix = default_arch_prefix.result
+.//-- 010:20140410 Add End (saitou) 
     .include "${te_file.arc_path}/t.class.extent.h"
   .else
     .invoke peis = PEIRenderInitializers( o_obj )
