Index: fx_domain_debug_utility.arc
===================================================================
--- fx_domain_debug_utility.arc	(revision 366)
+++ fx_domain_debug_utility.arc	(revision 367)
@@ -146,7 +146,7 @@
 .param inst_ref te_class	// te_class
   .//
   .assign attr_result = ""
-  .select many te_evts related by te_class->TE_SM[R2072]->TE_EVT[R2071]
+  .select many te_evts related by te_class->TE_SM[R2072]->TE_EVT[R2071] where ( selected.Used )
   .if( not_empty te_evts )
     .assign attr_result = ( "${te_c.Name}" + "_" ) + ( "${te_class.Key_Lett}" + "_EventDetail" )
   .end if
