Index: q.class.link.arc
===================================================================
--- q.class.link.arc	(revision 373)
+++ q.class.link.arc	(revision 374)
@@ -1009,6 +1009,7 @@
             .if ( aoth.Mult == 1 )
               .invoke assr_cast_class = GetObjectClassName( assr_obj )
               .assign form_name = "${form_name}.head"
+    if( ${form_name} )
               .assign aoth_name = "((${assr_cast_class.result}*)(${form_name}->next->object))->${aoth_member_name.result}"
               .assign while_exit="${form_name}->next"
               .assign form_name="${form_name}->next->object"
@@ -1044,6 +1045,7 @@
                 .//
                 .if (aone.Mult == 1)
                   .assign form_name = "${form_name}.head"
+    if( ${form_name} )
                   .assign while_exit="${form_name}->next"
                   .assign form_name="${form_name}->next->object"
                   .assign part_name="((${assr_cast_class.result}*)(${form_name}))->${aone_member_name.result}"
@@ -1053,8 +1055,8 @@
           .end if .// if ( is_this_obj_assr )
           .if ( is_gen )
             .if ( while_exit != "" )
-    while( ${while_exit} )
-        ${unrelate_method.result}( ${part_name} , ${aoth_name} , ${form_name} );
+      while( ${while_exit} )
+          ${unrelate_method.result}( ${part_name} , ${aoth_name} , ${form_name} );
             .else
     ${unrelate_method.result}( ${part_name} , ${aoth_name} , ${form_name} );
             .end if .// if ( while_exit != "" )
