Index: q.class.link.arc
===================================================================
--- q.class.link.arc	(revision 378)
+++ q.class.link.arc	(revision 384)
@@ -916,7 +916,10 @@
       .else
         .assign member_data_name.result = member_data_name.result + ".head"
     if ( ${te_relstore.self_name}->${member_data_name.result} )
-      while( ${te_relstore.self_name}->${member_data_name.result}->next )
+ .//-- 029:20140423 Modified Start (nomura)
+      .//while( ${te_relstore.self_name}->${member_data_name.result}->next )
+      while( ${te_relstore.self_name}->${member_data_name.result} )
+ .//-- 029:20140423 Modified End (nomura)
         .assign member_data_name.result = member_data_name.result + "->object"
         .assign isWhile = true
         .if ( te_relinfo.is_formalizer == TRUE )
