Index: q.domain.sync.arc
===================================================================
--- q.domain.sync.arc	(revision 385)
+++ q.domain.sync.arc	(revision 388)
@@ -113,7 +113,6 @@
 	  .end if
 	  .if ( included == FALSE) 
 	    .if (( te_dt.Include_File != "") and ("${te_dt.Include_File}" != "${te_file.types}.${te_file.hdr_file_ext}" ) )
-	    .print  "+++++ #include ${te_dt.Include_File}"
 #include "${te_dt.Include_File}" /* ${te_dt.Name} */
             .end if
           .end if
