Index: arc/fx_domain_debug_utility.arc
===================================================================
--- arc/fx_domain_debug_utility.arc	(revision 434)
+++ arc/fx_domain_debug_utility.arc	(revision 444)
@@ -210,7 +210,7 @@
   ${te_set.iterator_class_name} ite;
   int instNo = ${te_set.element_count}(&${class_ext.result}.active);
   
-  printf("Class Set Recovery Unit Total Instance No = %d\n",instNo);
+  printf("Class [${te_class.Name}] Total Instance Num = %d\n",instNo);
   ${te_set.iterator_reset}(&ite,&${class_ext.result}.active);
   while ( (ptr=(${te_class.GeneratedName}*)(${te_set.iterator_next}(&ite))) != 0 ){
     printf("[Addr] %p\n", ptr);
